blockdag/validate.go

```diff
@@ -471,105 +471,164 @@ func (dag *BlockDAG) checkBlockSanity(block *util.Block, flags BehaviorFlags) (t
 	if err != nil {
 		return 0, err
 	}
+	err = dag.checkBlockContainsAtLeastOneTransaction(msgBlock)
+	if err != nil {
+		return 0, err
+	}
+	err = dag.checkBlockContainsLessThanMaxBlockMassTransactions(msgBlock)
+	if err != nil {
+		return 0, err
+	}
+	transactions := block.Transactions()
```

Comment:
I think it would be nice, if similar to transactions, you also have header, and functions that only care about header receive it.

Reply:
We've spoken about this and decided to leave it as is.